### PR TITLE
feat(widget-builder): Add display type dropdown

### LINKED ISSUE #81505: Add type field
This is referencing the visualization type

### PR DESCRIPTION
**Pull request by nikkikapadia**
Added the Display Type dropdown to the slideout. Changing the type also updates the URL and is integrated with the widget builder context. It looks like this:

image

Closes https://github.com/getsentry/sentry/issues/81505

### CODE REVIEW

File: static/app/views/dashboards/widgetBuilder/components/datasetSelector.tsx [hunk 1]
```diff
@@ -81,6 +81,6 @@ const FeatureBadgeAlignmentWrapper = styled('div')`
 
 const Header = styled('h6')`
   font-size: ${p => p.theme.fontSizeLarge};
+  ${p => p.theme.tooltipUnderline()};
```
Comment (narsaynorath) [inline static/app/views/dashboards/widgetBuilder/components/datasetSelector.tsx:84]: Oh wait, I remembered, there's a `showUnderline` prop you can pass to `Tooltip`, does that remove this styling?

File: static/app/views/dashboards/widgetBuilder/components/newWidgetBuilder.spec.tsx [hunk 2]
```diff
@@ -177,4 +186,34 @@ describe('NewWidgetBuiler', function () {
       })
     );
   });
+
+  it('changes the visualization type', async function () {
```
Comment (narsaynorath) [inline static/app/views/dashboards/widgetBuilder/components/newWidgetBuilder.spec.tsx:190]: Long term, I'd actual prefer if we didn't test individual field changes in the builder as much. I'm still trying to find out the best way to test all of these but if the hook has coverage for the field changes and the fields have their own coverage, that means the widget builder tests shouldn't duplicate that coverage and test the logic that is specifically unique to that component. This should make it easier to test and also faster

File: static/app/views/dashboards/widgetBuilder/components/typeSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+export default WidgetBuilderTypeSelector;
+
+const SelectionWrapper = styled('div')`
+  display: flex;
+  gap: ${space(1)};
+`;
+
+const Header = styled('h6')`
+  font-size: ${p => p.theme.fontSizeLarge};
+  ${p => p.theme.tooltipUnderline()};
```
Comment (narsaynorath) [inline static/app/views/dashboards/widgetBuilder/components/typeSelector.tsx:84]: Same comment here about `showUnderline`

**Review by narsaynorath — APPROVED** [review]
Looking good!

File: static/app/views/dashboards/widgetBuilder/components/datasetSelector.tsx (hunk 1 above)
Comment (nikkikapadia) [reply static/app/views/dashboards/widgetBuilder/components/datasetSelector.tsx:84]: ahh good catch I didn't know that was a thing! and it does the same styling 👍

**Pull request merged**